core/drivers/pl022_spi.c

```diff
@@ -356,6 +289,21 @@ static inline uint32_t pl022_calc_freq(struct pl022_data *pd,
 	return pd->clk_hz / (cpsdvr * (1 + scr));
 }
 
+static void pl022_control_cs(struct spi_chip *chip, enum gpio_level value)
+{
+	struct pl022_data *pd = container_of(chip, struct pl022_data, chip);
+
+	if (pd->b_cs_callback && pd->cs_data.cs_cb) {
+		pd->cs_data.cs_cb(value);
+	} else if (!pd->b_cs_callback) {
+		while (read8(pd->base + SSPSR) & SSPSR_BSY)
+			DMSG("pl022 still transceiving - do NOT set CS yet!");
```

Comment:
This is a case where SW is faster than the IP HW. If we set the pinmux function for the CS pin as SPI, then the IP HW will just control the CS pin and there's no problem there, but if we set the function as GPIO, then this driver is responsible for controlling/pulsing/changing the CS signal. In this case, if we just change the signal without checking to see if the IP HW is busy (SSPSR_BSY), i.e. still transmitting/receiving, then we will cancel the transfer in the middle of transmission. The debug message lets us know that program execution is at the point where it's ready to change the CS signal, but the HW IP is still busy so we need to wait.